TombRaiderIII/Autosplitter.cs

```diff
@@ -0,0 +1,190 @@
+        public Autosplitter() => GameMemory.OnGameFound += Settings.SetGameVersion;
+
+        /// <summary>
+        ///     Determines the IGT.
+        /// </summary>
+        /// <param name="state"><see cref="LiveSplitState"/> passed by LiveSplit</param>
+        /// <returns>IGT as a <see cref="TimeSpan"/> if available, otherwise <see langword="null"/></returns>
+        public TimeSpan? GetGameTime(LiveSplitState state)
+        {
+            uint currentLevelTicks = GameMemory.Data.LevelTime.Current;
+            double currentLevelTime = (double)currentLevelTicks / IgtTicksPerSecond;
+
+            bool noLevelsCompleted = _completedLevels.Count != 0;
+            return noLevelsCompleted 
+                ? TimeSpan.FromSeconds(currentLevelTime)
+                : TimeSpan.FromSeconds(currentLevelTime + SumCompletedLevelTimes());
+        }
+
+        /// <summary>
+        ///     Sums completed levels' times.
+        /// </summary>
+        /// <returns>The sum of completed levels' times as a double.</returns>
+        /// <remarks>
+        ///     Because TR3's level order is variable and we must maintain compatibility with Section and NG+ runs,
+        ///     we read addresses based on <c>_completedLevels</c>.
+        /// </remarks>
+        private double SumCompletedLevelTimes()
+        {
+            var finishedLevelsTicks = 0;
+            // Add up the level times of interest from the game's memory.
+            foreach (Level completedLevel in _completedLevels)
+            {
+                int levelOffset = ((int)completedLevel - 1) * 0x2c; // - 1 because FirstLevelTimeAddress is already on the first address
+                var levelAddress = (IntPtr)(GameData.FirstLevelTimeAddress + levelOffset);
+                finishedLevelsTicks += GameMemory.Game.ReadValue<int>(levelAddress);
+            }
+            return (double)finishedLevelsTicks / IgtTicksPerSecond;
+        }
+
+        /// <summary>
+        ///     Determines if IGT should be paused when the game is quit or <see cref="GetGameTime"/> returns <see langword="null"/>
+        /// </summary>
+        /// <param name="state"><see cref="LiveSplitState"/> passed by LiveSplit</param>
+        /// <returns><see langword="true"/> if IGT should be paused, <see langword="false"/> otherwise</returns>
+        public bool IsGameTimePaused(LiveSplitState state) => true;
+
+        /// <summary>
+        ///     Determines if the timer should split.
+        /// </summary>
+        /// <param name="state"><see cref="LiveSplitState"/> passed by LiveSplit</param>
+        /// <returns><see langword="true"/> if the timer should split, <see langword="false"/> otherwise</returns>
+        public bool ShouldSplit(LiveSplitState state)
+        {
+            // Determine if the player is on the correct level to split; if not, we stop.
+            Level currentLevel = GameMemory.Data.Level.Current;
+            bool onCorrectLevelToSplit = _completedLevels.Count == 0
```

Comment:
Change to `bool onCorrectLevelToSplit = !_completedLevels.Contains(currentLevel);`.